docs/csharp/roslyn-sdk/tutorials/how-to-write-csharp-analyzer-code-fix.md

```diff
@@ -0,0 +1,615 @@
+
+The structure of the unit tests is shown in the two example tests created by the template. Open **MakeConstUnitTests.cs**. The first test, `TestMethod1` shows the structure of a test when you want to verify that analyzed code does not trigger a diagnostic. `TestMethod2` shows the structure of a test when analyzed code triggers a diagnostic and provides a fix.
+
+Change `TestMethod1` to the following code:
+
+[!code-csharp[Don't report diagnostics on const declarations](~/samples/csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst/MakeConst.Test/MakeConstUnitTests.cs#AlreadyConst "Test to ensure no diagnostics are reported on already const definition")]
+
+Run all the unit tests now. This new test passes. After all, your analyzer doesn't do anything yet. Add the following code to `AnalyzeNode` to ensure this test continues to pass:
+
+```csharp
+// Only consider local variable declarations that aren't already const.
+if (localDeclaration.Modifiers.Any(SyntaxKind.ConstKeyword))
+{
+    return;
+}
+```
+
+### Examine declaration initializers
+
+Constant declarations are not the only case where your analyzer can exit quickly. Any variable declaration that doesn't contain an initializer cannot be made constant. Also, any variable declaration whose initializer is not a compile-time constant cannot be made const. Write two new tests that verify these two rules:
+
+[!code-csharp[Don't report diagnostics on declarations that can't be const](~/samples/csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst/MakeConst.Test/MakeConstUnitTests.cs#CantBeConst "Test to ensure no diagnostics are reported when the declaration can't be made const")]
+
+There's one additional C# spec rule that you should take care of with the same code: If a declaration statement includes multiple variables, they must all be const, or none of them can be const. That means one other test to ensure that a diagnostic is not reported when a declaration statement has multiple variables, and not all of them can be made const. That suggests one more test:
+
+[!code-csharp[Don't report diagnostics on multiple declarations if all can't be const](~/samples/csharp/roslyn-sdk/Tutorials/MakeConst/MakeConst/MakeConst.Test/MakeConstUnitTests.cs#MultipleDeclarations "Test to ensure no diagnostics are reported when multiple declarations can't all be made const")]
+
+As before, these tests will pass now. It's time to write the code that checks for these conditions. You'll perform some semantic analysis using the <xref:Microsoft.CodeAnalysis.Diagnostics.SyntaxNodeAnalysisContext>. You use the `context` argument to determine whether the local variable declaration can be made `const`. A <xref:Microsoft.CodeAnalysis.SemanticModel?displayProperty=nameWithType> represents of all semantic information in a single source file. You can earn more in the article that covers [semantic models](../work-with-semantics.md). Add the following code to the `AnalyzeNode` method:
```

Comment:
1. You wrote `you can earn more...` but you meant `you can learn more...`
2. Again watch out for that `context` variable as mine was named `obj`